src/System.CommandLine/RootCommand.cs

```diff
@@ -33,9 +33,14 @@ public override string Name
             set => throw new NotSupportedException("The root command's name cannot be changed.");
         }
 
+        private static readonly Lazy<string> executablePath = 
+            new Lazy<string>(() => Assembly.GetEntryAssembly().Location);
+  
         private static readonly Lazy<string> executableName =
-            new Lazy<string>(() => Path.GetFileNameWithoutExtension(Assembly.GetEntryAssembly().Location));
+            new Lazy<string>(() => Path.GetFileNameWithoutExtension(executablePath.Value));
```

Comment:
So what is the point of these `Lazy<string>` here? Because of the static readonly properties below, these get evaluated during the static constructor, effectively mitigating the delayed evaluation.

Reply:
Good catch. I've changed them to be late-evaluated properties.